src/api.ts

```diff
@@ -10,9 +10,11 @@ import {
 
 /* Consumers that wish to define their own behavior may compose sets of these decorator factories together.
  */
-export function composeDecoratorFactories(
-    factories: DTODecoratorFactories[],
-): DTODecoratorFactories {
+export function composeDecoratorFactories<
+    CustomDTODecoratorFactories extends DTODecoratorFactories = DTODecoratorFactories,
+>(
+    factories: (DTODecoratorFactories | CustomDTODecoratorFactories)[],
```

Comment:
```suggestion
    factories: (FactoriesType)[],
```

Unless I'm missing something, we should only have the parameterized type here, not the base as well.